packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -3478,3 +3480,39 @@ class _InputDateRangePickerState extends State<_InputDateRangePicker> {
     );
   }
 }
+ // A text input formatter that formats the input as a date in MM/DD/YYYY format
```

Comment:
Instead of only the MMDDYYYY format, the formatter should probably consider other formats that can be found from i18n.